homeassistant/components/emoncms/sensor.py

```diff
@@ -107,6 +106,12 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         if sensor_names is not None:
             name = sensor_names.get(int(elem["id"]), None)
 
+        unit = elem.get("unit")
+        if unit:
+            unit_of_measurement = unit
+        else:
+            unit_of_measurement = config.get(CONF_UNIT_OF_MEASUREMENT)
```

Comment:
We could even make this line easier by changing the variable in line 77 from `unit_of_measurement` to `config_unit` and then using it here. This would make things easier since we do not need to access the config multiple times. Unfortunately, I cannot change line 77 from github so you will have to rename it 😄
```suggestion
            unit_of_measurement = config_unit
```

Reply:
OK done (missed your last comment!).